sourcetracker/_sourcetracker.py

```diff
@@ -949,40 +765,105 @@ def _gibbs(source_df, sink_df, alpha1, alpha2, beta, restarts,
     >>> burnin = 2
     >>> delay = 2
 
-    Call without a cluster
-    >>> mp, mps = _gibbs(source_df, sink_df, alpha1, alpha2, beta, restarts,
-                         draws_per_restart, burnin, delay)
+    # Call without a cluster
+    >>> mpm, mps, fas = _gibbs(source_df, sink_df, alpha1, alpha2, beta,
+                               restarts, draws_per_restart, burnin, delay)
 
-    Start a cluster and call the function.
+    # Start a cluster and call the function.
     >>> jobs = 4
     >>> subprocess.Popen('ipcluster start -n %s --quiet' % jobs, shell=True)
     >>> time.sleep(25)
     >>> c = Client()
-    >>> mp, mps = _gibbs(source_df, sink_df, alpha1, alpha2, beta, restarts,
-                         draws_per_restart, burnin, delay, cluster=c)
+    >>> mpm, mps, fas = _gibbs(source_df, sink_df, alpha1, alpha2, beta,
+                               restarts, draws_per_restart, burnin, delay,
+                               cluster=c)
+    '''
+    cp = ConditionalProbability(alpha1, alpha2, beta, sources.values)
+    f = partial(gibbs_sampler, cp=cp, restarts=restarts,
+                draws_per_restart=draws_per_restart, burnin=burnin,
+                delay=delay)
+    if cluster is not None:
+        results = cluster[:].map(f, sinks.values, block=True)
+    else:
+        results = list(map(f, sinks.values))
+
+    mpm, mps, fas = collate_gibbs_results([i[0] for i in results],
+                                          [i[1] for i in results],
+                                          sources.index, sources.columns,
+                                          sinks.index)
+    return mpm, mps, fas
+
+
+def collate_gibbs_results(mixing_proportions, feature_assignments,
```

Comment:
You should name functions that will be private with a leading underscore. We'll ultimately remove the leading underscore from the public functions, once we're ready to start supporting those APIs.